homeassistant/components/glances/sensor.py

```diff
@@ -1,133 +1,58 @@
-        vol.Optional(CONF_USERNAME): cv.string,
-        vol.Optional(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_SSL, default=False): cv.boolean,
-        vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
-        vol.Optional(CONF_RESOURCES, default=["disk_use"]): vol.All(
-            cv.ensure_list, [vol.In(SENSOR_TYPES)]
-        ),
-        vol.Optional(CONF_VERSION, default=DEFAULT_VERSION): vol.In([2, 3]),
-    }
-)
+_LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the Glances sensors."""
-    from glances_api import Glances
-
-    name = config[CONF_NAME]
-    host = config[CONF_HOST]
-    port = config[CONF_PORT]
-    version = config[CONF_VERSION]
-    var_conf = config[CONF_RESOURCES]
-    username = config.get(CONF_USERNAME)
-    password = config.get(CONF_PASSWORD)
-    ssl = config[CONF_SSL]
-    verify_ssl = config[CONF_VERIFY_SSL]
-
-    session = async_get_clientsession(hass, verify_ssl)
-    glances = GlancesData(
-        Glances(
-            hass.loop,
-            session,
-            host=host,
-            port=port,
-            version=version,
-            username=username,
-            password=password,
-            ssl=ssl,
-        )
-    )
+    """Set up the Glances sensors is done through async_setup_entry."""
+    return True
 
-    await glances.async_update()
 
-    if glances.api.data is None:
-        raise PlatformNotReady
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Glances sensors."""
 
+    glances_api = hass.data[DOMAIN][config_entry.entry_id]
+    client_name = config_entry.data[CONF_NAME]
     dev = []
-    for resource in var_conf:
-        dev.append(GlancesSensor(glances, name, resource))
+    for sensor_type in SENSOR_TYPES:
+        dev.append(
+            GlancesSensor(
+                glances_api, client_name, SENSOR_TYPES[sensor_type][0], sensor_type
+            )
+        )
 
     async_add_entities(dev, True)
 
 
 class GlancesSensor(Entity):
     """Implementation of a Glances sensor."""
 
-    def __init__(self, glances, name, sensor_type):
+    def __init__(self, glances_api, client_name, sensor_name, sensor_type):
         """Initialize the sensor."""
-        self.glances = glances
-        self._name = name
+        self.glances_api = glances_api
+        self._name = sensor_name
+        self.client_name = client_name
         self.type = sensor_type
         self._state = None
         self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return "{} {}".format(self._name, SENSOR_TYPES[self.type][0])
+        return f"{self.client_name} {self._name}"
+
+    @property
+    def unique_id(self):
+        """Set unique_id for sensor."""
+        return f"{self.glances_api.host}-{self.name}"
```

Comment:
What is the host value? Is it an ip address? We don't allow ip address for unique_id. IP addresses can change and the unique_id must be stable.

https://developers.home-assistant.io/docs/en/entity_registry_index.html#unique-id-requirements

Reply:
What you are talking about is true for devices. But in this case I only want to link the entities to the integration. Glances doesn't offer mac or serial. How can we set the unique ID in this case?

Reply:
If it's ok to lose all entity settings when removing the config entry and creating a new entry but after a host change, I guess it's ok.